core/src/main/java/io/grpc/internal/DelayedClientTransport2.java

```diff
@@ -85,13 +87,50 @@
   @Nullable
   private SubchannelPicker lastPicker;
 
-  DelayedClientTransport2(Executor streamCreationExecutor) {
-    this.streamCreationExecutor = streamCreationExecutor;
+  @GuardedBy("lock")
+  private long lastPickerVersion;
+
+  /**
+   * Creates a new delayed transport.
+   *
+   * @param defaultAppExecutor pending streams will create real streams and run bufferred operations
+   *        in an application executor, which will be this executor, unless there is on provided in
+   *        {@link CallOptions}.
+   * @param channelExecutor all listener callbacks of the delayed transport will be run from this
```

Comment:
I was a bit surprised to see the channel executor provided to this class (instead of just creating a new private one), but I can believe there may be benefits. Maybe pass the instance in during `start()` to make the association with the `listener` more obvious, especially for the consumer of this class?